### PR TITLE
Support for tidb.

### PR DESCRIPTION
**Pull request by wshwsh12**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,5 @@ benchmark/*/queries-monetdb/*.ret
 benchmark/*/queries-vectorwise/*.out
 benchmark/*/queries-vectorwise/*.err
 benchmark/*/queries-vectorwise/*.ret
+
+/var
```
Comment (zz-jason) [inline .gitignore]: not needed?

File: README.md [hunk 2]
```diff
@@ -1,35 +1,80 @@
-# Public BI benchmark
+# Public BI benchmark for TiDB
 
-User generated benchmark derived from the DBTest'18 paper [1] by Tableau. It contains real data and queries from 46 public workbooks in Tableau Public [2].
+## 数据集大小
```
Comment (zz-jason) [inline README.md]: Please use English. You can write these introduction in the Chinese version.

File: README.md [hunk 3]
```diff
@@ -1,35 +1,80 @@
...
-Each directory is associated with a workbook and contains:
+## 使用方法
 ```
-samples:        a sample of each .csv file (first 20 rows)
-tables:         .sql files containing the schema of each .csv file
-queries:        .sql files containing the queries
-data-urls.txt:  links for downloading the full .csv.bz2 compressed files
+go run main.go [download|prepare|run|cleanup]: 
+  -P int
+    	MySQL Port (default 3306)
```
Comment (zz-jason) [inline README.md]: How about setting the default port to 4000?

File: README.md [hunk 4]
```diff
@@ -1,35 +1,80 @@
...
+    	input benchmark resource (default "./benchmark")
+  -ignore-run-error
+    	Ignore errors when run
+  -o string
+    	output directory to save cvs files and logs (default "./var")
+  -p string
+    	MySQL Password
+  -sample
+    	Use sample to test
+  -u string
```
Comment (zz-jason) [inline README.md]: How about reorg the options like this:
1. options used to connect the database server.
2. options used to control the test behavior.

File: README.md [hunk 5]
```diff
@@ -1,35 +1,82 @@
...
+    	Use sample to test
+  -ignore-run-error
+    	Ignore errors when running
 ```
+Hint：
 
-There are 46 workbooks containing 206 tables (.csv files) with the total size of 41 GB compressed and 386 GB uncompressed.
+1. When using sample to test, you can run `prepare` directly without `download` .
+
+2. When using datasets to test, I recommend to download datasets in other ways first, and use options `-o` to select file directory. ( Maybe the download speed is too slow, VPN or downloading from internal server can work well.)
```
Comment (zz-jason) [inline README.md]:
```suggestion
2. When using datasets to test, download datasets in other ways first, use options `-o` to select file directory. (Maybe the download speed is too slow, VPN or downloading from internal server can work well.) 
```

File: README.md [hunk 6]
```diff
@@ -1,35 +1,82 @@
...
-There are 46 workbooks containing 206 tables (.csv files) with the total size of 41 GB compressed and 386 GB uncompressed.
+1. When using sample to test, you can run `prepare` directly without `download` .
+
+2. When using datasets to test, I recommend to download datasets in other ways first, and use options `-o` to select file directory. ( Maybe the download speed is too slow, VPN or downloading from internal server can work well.) 
 
-Multiple .csv files may overlap but are not identical. This is because Tableau extracts the same workbook in multiple different ways for different queries.
+## Status
 
-The benchmark contains 646 queries which have been tested on MonetDB. The same queries with VectorWise specific syntax are available on the `dev/master` branch under `benchmark/<workbook>/queries-vectorwise`.
+For running the bench in TiDB, I have adjusted some datasets and SQL queries. The adjustment is as follows.
```
Comment (zz-jason) [inline README.md]:
```suggestion
For running the benchmark in TiDB, some datasets and SQL queries are adjusted. The adjustment is as follows.
```

File: README.md [hunk 7]
```diff
@@ -1,35 +1,82 @@
...
 
-## dev/master branch
+### Delate datasets:
+'Wins' and 'USCensus' （The number of columns is greater than 512.）
 
-The `dev/master` branch contains:
-- scripts to download and decompress the entire dataset
-- scripts to create tables (and adapt schema syntax), load data and run queries for MonetDB and VectorWise
-- scripts that document the process of cleaning, fixing and adapting the data and queries from the original Tableau specific form
+In workbook 'Wins' , there are 13 queries.
```
Comment (zz-jason) [inline README.md]:
```
Datasets 'Wins' and 'USCensus' （The number of columns is greater than 512.）are deleted.
* In workbook 'Wins' , there are 13 queries. 
* In workbook 'USCensus' , there are 8 queries.
```

File: README.md [hunk 8]
```diff
@@ -1,35 +1,82 @@
...
-For more info see the README on `dev/master`.
+In workbook 'USCensus' , there are 8 queries.
 
-## References
+### Adjust SQL queries:
+Change `cast xx as bigint` to `cast xx as signed`
 
-[1] Vogelsgesang, Adrian, et al. "Get real: How benchmarks fail to represent the real world." Proceedings of the Workshop on Testing Database Systems. ACM, 2018.\
-[2] https://public.tableau.com
+### Delate SQL queries:
```
Comment (zz-jason) [inline README.md]:
```suggestion
### Deleted SQL queries:
```

**Review by zz-jason — COMMENTED** [review]
LGTM

**Comment by zhouqiang-cl** [comment]
LGTM

**Pull request merged**